graphix/gflow.py

```diff
@@ -245,123 +140,71 @@ def find_flow(
+    of l_k from 1), and for each measurements of qubit i we must perform corrections on
+    qubits in g(i), depending on the measurement outcome.
 
-    Ref: Mhalla and Perdrix, International Colloquium on Automata,
-    Languages, and Programming (Springer, 2008), pp. 857-868.
+    For more details of gflow, see Browne et al., NJP 9, 250 (2007).
+    We use the extended gflow finding algorithm in Backens et al., Quantum 5, 421 (2021).
 
     Parameters
     ----------
-    nodes: set
-        labels of all qubits (nodes)
-    edges: set
-        edges
-    iset: set
-        set of node labels for input
-    oset: set
-        set of node labels for output
-    v_c: set
-        correction candidate qubits
-    f: dict
-        flow function. f[i] is the qubit to be measured after qubit i.
-    l_k: dict
-        layers obtained by flow algorithm. l_k[d] is a node set of depth d.
-    k: int
-        current layer number.
-    meas_planes: dict
-        measurement planes for each qubits. meas_planes[i] is the measurement plane for qubit i.
+    graph : `networkx.Graph`
+        The underlying graph.
+    iset : `collections.abc.Set`
+        Input nodes.
+    oset : `collections.abc.Set`
+        Output nodes.
+    meas : `collections.abc.Mapping`, optional
+        Measurement specs for each qubit, by default all XY.
 
-    Outputs
+    Returns
     -------
-    f: list of nodes
-        causal flow function. f[i] is the qubit to be measured after qubit i.
-    l_k: dict
-        layers obtained by gflow algorithm. l_k[d] is a node set of depth d.
+    Gflow function and layer if found, otherwise `None`.
     """
-    v_out_prime = set()
-    c_prime = set()
-
-    for q in v_c:
-        nb = search_neighbor(q, edges)
-        p_set = nb & (nodes - oset)
-        if len(p_set) == 1:
-            # Iterate over p_set assuming there is only one element p
-            (p,) = p_set
-            f[p] = {q}
-            l_k[p] = k
-            v_out_prime |= {p}
-            c_prime |= {q}
-    # determine whether there exists flow
-    if not v_out_prime:
-        if oset == nodes:
-            return f, l_k
-        return None, None
-    return flowaux(
-        nodes,
-        edges,
-        iset,
-        oset | v_out_prime,
-        (v_c - c_prime) | (v_out_prime & (nodes - iset)),
-        f,
-        l_k,
-        k + 1,
-    )
+    if meas is None:
+        meas = _default_construct(graph.nodes - oset)
+    return gflow_module.find(graph, iset, oset, {k: _p_convert(v) for k, v in meas.items()})
 
 
 def find_pauliflow(
-    graph: nx.Graph,
-    iset: set[int],
-    oset: set[int],
-    meas_planes: dict[int, Plane],
-    meas_angles: dict[int, float],
-    mode: str = "single",
-) -> tuple[dict[int, set[int]], dict[int, int]]:
+    graph: nx.Graph[int],
+    iset: AbstractSet[int],
+    oset: AbstractSet[int],
+    meas: Mapping[int, AnyMeasurement] | None = None,
```

Comment:
The type we want here is something isomorphic to `Mapping[int, Plane | Axis]`, and to perform the conversion from `Measurement` in `generate_from_graph`.